client/src/app/core/services/cache.service.ts

```diff
@@ -1,21 +1,151 @@
 import { Injectable } from '@angular/core';
+import { LocalStorage } from '@ngx-pwa/local-storage';
+import { Observable } from 'rxjs';
 
 /**
- * Handles all incoming and outgoing notify messages via {@link WebsocketService}.
+ * Container objects for the setQueue.
+ */
+interface SetContainer {
+    key: string;
+    item: any;
+    callback: (value: boolean) => void;
+}
+
+/**
+ * Container objects for the removeQueue.
+ */
+interface RemoveContainer {
+    key: string;
+    callback: (value: boolean) => void;
+}
+
+/**
+ * Provides an async API to an key-value store using ngx-pwa which is internally
+ * using IndexedDB or localStorage as a fallback.
  */
 @Injectable({
     providedIn: 'root'
 })
 export class CacheService {
     /**
-     * Constructor to create the NotifyService. Registers itself to the WebsocketService.
-     * @param websocketService
+     * The queue of waiting set requests. Just one request (with the same key, which is
+     * an often case) at the time can be handeled. The SetContainer encapsulates the key,
+     * item and callback.
+     */
+    private _setQueue: SetContainer[] = [];
```

Comment:
`private _` Should be avoided according to angular styleguides.

https://angular.io/guide/styleguide#style-03-04

I use `private _myVar` myself if I add `get myVar()` and/or `set myVar(new: any)` in a class, but would advise to avoid it in any other case.